pkg/utils/validation/apigroups/apigroups.go

```diff
@@ -0,0 +1,352 @@
+		"storage.k8s.io/v1alpha1/csistoragecapacities":                            {},
+		"storage.k8s.io/v1alpha1/volumeattachments":                               {},
+		"storage.k8s.io/v1beta1/csidrivers":                                       {},
+		"storage.k8s.io/v1beta1/csinodes":                                         {},
+		"storage.k8s.io/v1beta1/csistoragecapacities":                             {},
+		"storage.k8s.io/v1beta1/storageclasses":                                   {},
+		"storage.k8s.io/v1beta1/volumeattachments":                                {},
+		"v1/componentstatuses":                                                    {},
+		"v1/configmaps":                                                           {},
+		"v1/endpoints":                                                            {},
+		"v1/events":                                                               {},
+		"v1/limitranges":                                                          {},
+		"v1/namespaces":                                                           {},
+		"v1/nodes":                                                                {},
+		"v1/persistentvolumeclaims":                                               {},
+		"v1/persistentvolumes":                                                    {},
+		"v1/pods":                                                                 {},
+		"v1/podtemplates":                                                         {},
+		"v1/replicationcontrollers":                                               {},
+		"v1/resourcequotas":                                                       {},
+		"v1/secrets":                                                              {},
+		"v1/serviceaccounts":                                                      {},
+		"v1/services":                                                             {},
+	}
+)
+
+// IsAPISupported returns true if the given API is supported for the given Kubernetes version.
+// An API is only supported if it's a known API and its version range contains the given Kubernetes version.
+func IsAPISupported(api, version string) (bool, string, error) {
+	var versionRange *APIVersionRange
+
+	apiGroupVersion, apiGVR, err := SplitAPI(api)
+	if err != nil {
+		return false, "", err
+	}
+
+	if apiGVR != "" {
+		vr, ok := apiGVRVersionRanges[apiGVR]
+		if !ok {
+			return false, "", fmt.Errorf("unknown API group version resource %q", apiGVR)
+		}
+		versionRange = vr
+	} else {
+		vr, ok := apiGroupVersionRanges[apiGroupVersion]
+		if !ok {
+			return false, "", fmt.Errorf("unknown API group version %q", apiGroupVersion)
+		}
+		versionRange = vr
+	}
+
+	contains, err := versionRange.Contains(version)
+	return contains, versionRange.SupportedVersionRange(), err
+}
+
+// APIVersionRange represents a version range of type [AddedInVersion, RemovedInVersion).
+type APIVersionRange struct {
+	Required              bool
+	RequiredForWorkerless bool
```

Comment:
So `Required` means "required for normal shoots, but not for workerless shoots", right? If an API is required for both shoot types, both fields must be set to `true`? If so, maybe make this more clear

Reply:
Yes.